### DISCUSSION TITLE
Some targets stay outdated after running `tar_make()`

### DISCUSSION THREAD

**Discussion by JohannesNE** [ropensci/targets#671] [Help] [answered]
After running `tar_make()` my network looks like this:

![image](https://user-images.githubusercontent.com/11048760/138271026-5ffca691-34f2-4b26-9294-5d0708cd5cad.png)

```
tar_outdated()
# [1] "beats_vent_index"     "qrs"                  "vent_settings_events"
# [4] "subjects"             "subject_files"        "beats"               
# [7] "subject_files_files"
```

If I run `tar_make()` again, everything is skipped, but the targets stay "outdated".

I make the target for the files with `tar_files(subject_files, list.files("path/to/files/", full.names = TRUE))`.

**Comment by wlandau** [comment, marked as answer]
`tar_files()` works by defining an upstream target to decide on the file paths and a downstream target to track them. The upstream target needs `tar_cue(mode = "always")`, which forces it and its downstream neighbors to show as outdated in the graph even though those targets may be skipped. This is a limitation. Please have a look at `tar_files_input()`, it should not have this issue.

**Reply by JohannesNE** [reply]
Thanks, that worked! So `tar_files()` are for file outputs?

**Reply by wlandau** [reply]
Yes, the files of `tar_files()` are resolved dynamically when the pipeline runs, whereas the files in `tar_files_input()` are resolved when you define the target.
